src/analysis/explicitly_annotated_unittests.d

```diff
@@ -0,0 +1,116 @@
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+module analysis.explicitly_annotated_unittests;
+
+import dparse.lexer;
+import dparse.ast;
+import analysis.base : BaseAnalyzer;
+
+import std.stdio;
+
+/**
+ * Requires unittests to be explicitly annotated with either @safe or @system
+ */
+class ExplicitlyAnnotatedUnittestCheck : BaseAnalyzer
+{
+	enum string KEY = "dscanner.style.explicitly_annotated_unittest";
+	enum string MESSAGE = "A unittest should be annotated with at least @safe or @system";
+
+	///
+	this(string fileName, bool skipTests = false)
+	{
+		super(fileName, null, skipTests);
+	}
+
+	override void visit(const Declaration decl)
+	{
+		if (decl.unittest_ !is null)
+		{
+			bool isSafeOrSystem = false;
+			if (decl.attributes !is null)
+			foreach (attribute; decl.attributes)
+			{
+				if (attribute.atAttribute !is null)
+				{
+					auto token = attribute.atAttribute.identifier.text;
+					if (token == "safe" || token == "system")
+					{
+						isSafeOrSystem = true;
+						break;
+					}
+				}
+			}
+			if (!isSafeOrSystem)
+				addErrorMessage(decl.unittest_.line, decl.unittest_.column, KEY, MESSAGE);
+		}
+		decl.accept(this);
+	}
+
+	alias visit = BaseAnalyzer.visit;
+
+}
+
+unittest
+{
+	import std.stdio : stderr;
+	import std.format : format;
+	import analysis.config : StaticAnalysisConfig, Check;
+	import analysis.helpers : assertAnalyzerWarnings;
+
+	StaticAnalysisConfig sac;
+	sac.explicitly_annotated_unittests = Check.enabled;
+
+	assertAnalyzerWarnings(q{
+		@safe unittest {
```

Comment:
can you change all the test to one line  `... unittest {}` ? (and also in the nested tests)